internal/module/services/tag_manager.go

```diff
@@ -0,0 +1,154 @@
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+
+	//"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+	"github.com/aws/aws-sdk-go/service/dynamodb/expression"
+	//"github.com/aws/aws-sdk-go/service/dynamodb/expression"
+)
+
+const (
+	DefaultTagTableName       = "terrarium-module-tags"
+	DefaultTagManagerEndpoint = "tag_manager:3001"
+)
+
+var (
+	TagTableName                      string = DefaultTagTableName
+	TagManagerEndpoint                string = DefaultTagManagerEndpoint
+	TagPublished                             = &terrarium.Response{Message: "Tag published."}
+	ModuleTagTableInitializationError        = status.Error(codes.Unknown, "Failed to initialize table for tags.")
+	//MarshalModuleVersionError              = status.Error(codes.Unknown, "Failed to marshal module version.")
+	PublishModuleTagError    = status.Error(codes.Unknown, "Failed to publish module tag.")
+	ConnectToTagManagerError = status.Error(codes.Unknown, "Failed to connect to TagManager service.")
+)
+
+type TagManagerService struct {
+	UnimplementedTagManagerServer
+	Db     dynamodbiface.DynamoDBAPI
+	Table  string
+	Schema *dynamodb.CreateTableInput
+}
+
+type ModuleTag struct {
+	Name       string   `json:"name" bson:"name" dynamodbav:"name"`
+	Tags       []string `json:"tags" bson:"tags" dynamodbav:"tags"`
+	CreatedOn  string   `json:"created_on" bson:"created_on" dynamodbav:"created_on"`
+	ModifiedOn string   `json:"modified_on" bson:"modified_on" dynamodbav:"modified_on"`
+}
+
+// RegisterWithServer registers TagManagerService with grpc server
+func (s *TagManagerService) RegisterWithServer(grpcServer grpc.ServiceRegistrar) error {
+	if err := storage.InitializeDynamoDb(s.Table, s.Schema, s.Db); err != nil {
+		log.Println(err)
+		return ModuleTagTableInitializationError
+	}
+
+	RegisterTagManagerServer(grpcServer, s)
+
+	return nil
+}
+
+func (s *RegistrarService) PublishTag(ctx context.Context, request *terrarium.PublishTagRequest) (*terrarium.Response, error) {
+	log.Println("Publish module tag.")
+
+	res, err := s.Db.GetItem(&dynamodb.GetItemInput{
+		TableName: aws.String(TagTableName),
+		Key: map[string]*dynamodb.AttributeValue{
+			"name": {S: aws.String(request.GetName())},
+		},
+	})
+
+	if err != nil {
+		log.Println(err)
+		return nil, ModuleGetError
+	}
+
+	if res.Item == nil {
+		ms := ModuleTag{
+			Name:       request.GetName(),
+			Tags:       request.GetTags(),
+			CreatedOn:  time.Now().UTC().String(),
+			ModifiedOn: time.Now().UTC().String(),
+		}
+
+		av, err := dynamodbattribute.MarshalMap(ms)
+
+		if err != nil {
+			log.Println(err)
+			return nil, MarshalModuleError
+		}
+
+		in := &dynamodb.PutItemInput{
+			Item:      av,
+			TableName: aws.String(RegistrarTableName),
```

Comment:
We should refer to the tags table as we're storing module tags in the tags table.  
```
TableName: aws.String(TagTableName)
```